src/FileIndexer.ts

```diff
@@ -608,6 +629,25 @@ export class FileIndexer {
 
     return this.checker.getTypeAtLocation(node)
   }
+
+  /**
+   * @param sym ts.Symbol with declarations to process.
+   * @returns an array of unique non-empty SCIP symbol values extracted
+   * from ts.Symbol declarations.
+   */
+  private uniqueDeclarationSymbolValues(sym?: ts.Symbol): string[] {
```

Comment:
nit: I don't have a strong opinion on this, but I've at least been trying to have things which don't require access to state be free functions (there are some examples at the end of this file). Makes it easier to understand which code is "pure" vs which code potentially modifies some state inside the indexer.

Reply:
@varungandhi-src, we call `this.scipSymbol(declaration)` in this method that uses the `FileIndexer` state.

Reply:
Oh sorry, my bad. 🤦🏽